src/ws/service/synapse_handlers.rs

```diff
@@ -0,0 +1,183 @@
+use std::sync::Arc;
```

Comment:
maybe this file should be under `ports/synapse`?

Reply:
Same. The content of the file is tightly coupled with the WebSocket Server logic and the `.proto` definitions.

Reply:
Applies here too
> Actually not that much, I think we could move it, but I'd like to re think the error handling of it all.